sdk/identity/Azure.Identity/README.md

```diff
@@ -1,99 +1,175 @@
-* Include this single line of links targeting your product's content at the bottom of the introduction, making any adjustments as necessary (for example, NuGet instead of PyPi):
+### Install the package
 
-  [Source code](https://github.com/Azure/azure-sdk-for-python/tree/master/azure-batch) | [Package (PyPi)](https://pypi.org/project/azure-batch/) | [API reference documentation](https://docs.microsoft.com/python/api/overview/azure/batch?view=azure-python) | [Product documentation](https://docs.microsoft.com/azure/batch/)
+Install the Azure Identity client library for .NET with [NuGet][nuget]:
 
-> TIP: Your README should be as **brief** as possible but **no more brief** than necessary to get a developer new to Azure, the service, or the package up and running quickly. Keep it brief, but include everything a developer needs to make their first API call successfully.
+```PowerShell
+Install-Package Azure.Identity -IncludePrerelease
+```
 
-## Getting started
+### Prerequisites
+* An [Azure subscription][azure_sub].
+* An existing Azure Active Directory service principal. If you need to create a service principal, you can use the Azure Portal or [Azure CLI][azure_cli].
+
+#### Creating a Service Principal with the Azure CLI
+Use the [Azure CLI][azure_cli] snippet below to create/get client secret credentials.
+
+ * Create a service principal and configure its access to Azure resources:
+    ```bash
+    az ad sp create-for-rbac -n <your-application-name> --skip-assignment
+    ```
+    Output:
+    ```json
+    {
+        "appId": "generated-app-ID",
+        "displayName": "dummy-app-name",
+        "name": "http://dummy-app-name",
+        "password": "random-password",
+        "tenant": "tenant-ID"
+    }
+    ```
+* Use the returned credentials above to set  **AZURE_CLIENT_ID**(appId), **AZURE_CLIENT_SECRET**(password) and **AZURE_TENANT_ID**(tenant) [environment variables](#environment-variables).
 
-This section should include everything a developer needs to do to install and create their first client connection *very quickly*.
+## Key concepts
+### Credentials
 
-### Install the package
+A credential is a class which contains or can obtain the data needed for a service client to authenticate requests. Service clients across Azure SDK accept credentials when they are constructed and use those credentials to authenticate requests to the service. Azure Identity offers a variety of credential classes in the `Azure.Identity` namespace capable of acquiring an AAD token. All of these credential classes are implementations of the `TokenCredential` abstract class in `Azure.Core`, and can be used by any service client which can be constructed with a `TokenCredential`. 
```

Comment:
It will be good to clarify what Azure.Core is or a link to the package or documentation as external users might not know about it